surrogate_gossiper.go

```diff
@@ -56,7 +57,12 @@ func (s *surrogateGossiper) OnGossip(update []byte) (GossipData, error) {
 	// (this time limit is arbitrary; surrogateGossiper should pass on new gossip immediately
 	// so there should be no reason for a duplicate to show up after a long time)
 	updateTime := now()
-	deleteBefore := updateTime.Add(-gossipInterval)
+	var deleteBefore time.Time
+	if s.router != nil && s.router.Config.GossipInterval != nil {
```

Comment:
This repeats the code in `actorLoop()` - can we pull it out to a method on `router` ?

Reply:
Done.